chain/network/src/routing/graph_v2/mod.rs

```diff
@@ -650,36 +650,34 @@ impl GraphV2 {
             edges: vec![],
         };
 
-        Self {
+        let inner = Arc::new(Mutex::new(Inner {
+            config,
+            edge_cache,
+            local_edges: HashMap::new(),
+            peer_distances: HashMap::new(),
+            my_distances: HashMap::from([(local_node, 0)]),
+            my_distance_vector,
+        }));
+
+        Arc::new_cyclic(|weak| Self {
+            inner: inner.clone(),
             routing_table: RoutingTableView::new(),
-            inner: Arc::new(Mutex::new(Inner {
-                config,
-                edge_cache,
-                local_edges: HashMap::new(),
-                peer_distances: HashMap::new(),
-                my_distances: HashMap::from([(local_node, 0)]),
-                my_distance_vector,
-            })),
             unreliable_peers: ArcSwap::default(),
-            runtime: Runtime::new(),
-        }
+            updater: new_owned_multithread_actor(1, {
```

Comment:
Feels weird to have a multithread actor with num_threads 1. I was wondering if we should have this ability to have owned actor handles for TokioRuntimeHandle instead of MultithreadRuntimeHandle?